### PR TITLE
Implement Access Token Check and Redirect

### LINKED ISSUE #122: Feature : Adding Sessions to the App
# Feature Request: Add Sessions to the App

## Summary  
Implement session management to maintain user state across requests, ensuring a seamless and secure experience.

## Expected Outcomes  
1. Users stay logged in across multiple requests.  
2. Unauthorized access after session expiry redirects to the login page.  
3. Logout terminates the session cleanly.

## Present Outcomes  
1. Users can access specific HTML files by entering the file name manually in the web search, which may lead to unauthorized access and lack of session management.

@Harshdev098 Pls assign me this task. Give me labels like hacktoberfest,gssoc-extd and level3.

### PR DESCRIPTION
**Pull request by ygowthamr**
**Description:**

Fix #122

- Added a check for the access token in `localStorage`.
- Implemented a redirect to the main page if the token is absent.
- Enhances security by preventing unauthorized access to protected routes.
- Improves user experience by guiding users to the main entry point.
- Changes are tested and ready for review.

@Harshdev098 
Pls add labels, gssoc-extd,level3,hacktoberfest-accepted lables.

### CODE REVIEW

File: public/choose-file.html [hunk 1]
```diff
@@ -100,6 +100,10 @@
 
   <div style="position: absolute; bottom: 30px; left: 30px;" class="gtranslate_wrapper"></div>
     <script>
+        const token = localStorage.getItem('accessToken');
+        if (!token) {
```
Comment (Harshdev098) [inline public/choose-file.html:104]: Don't add this in choose-file.html

File: public/choose-signup-file.html [hunk 2]
```diff
@@ -99,6 +99,10 @@
 <body>
   <div style="position: absolute; bottom: 30px; left: 30px;" class="gtranslate_wrapper"></div>
     <script>
+        const token = localStorage.getItem('accessToken');
+        if (!token) {
```
Comment (Harshdev098) [inline public/choose-signup-file.html:103]: Don't add this in choose-signup-file

File: public/dashboard.html [hunk 3]
```diff
@@ -243,7 +243,7 @@ <h3 style="margin-left:3vw;font-size: 22px;">Edit Profile</h3>
                 });
 
                 if (response.ok) {
-                    localStorage.removeItem('token'); // Remove the token from storage
+                    localStorage.removeItem('accessToken'); // Remove the token from storage
```
Comment (Harshdev098) [inline public/dashboard.html:246]: I think you have not implemented the token check in the dashboard file

**Review by Harshdev098 — CHANGES_REQUESTED** [review]
Also don't add token check in signup and login page

File: public/choose-file.html (hunk 1 above)
Comment (ygowthamr) [reply public/choose-file.html]: Done

File: public/choose-signup-file.html (hunk 2 above)
Comment (ygowthamr) [reply public/choose-signup-file.html]: done

File: public/dashboard.html (hunk 3 above)
Comment (ygowthamr) [reply public/dashboard.html:250]: done added token check

**Comment by ygowthamr** [comment]
@Harshdev098 
Updated required changes pls check and merge my PR

**Comment by ygowthamr** [comment]
@Harshdev098 
It is level3 issue pls assign me level3.

**Comment by ygowthamr** [comment]
@Harshdev098 
Pls change this pr level from level2 to level3.

**Comment by ygowthamr** [comment]
@Harshdev098 
Can you pls update the level to level3 before the gssoc leaderboard update.

**Comment by ygowthamr** [comment]
@Harshdev098 
Thank you

**Pull request merged**